pinot-broker/src/main/java/org/apache/pinot/broker/broker/helix/HelixBrokerStarter.java

```diff
@@ -117,13 +124,18 @@ public HelixBrokerStarter(PinotConfiguration brokerConf, String clusterName, Str
       brokerHost = _brokerConf.getProperty(CommonConstants.Helix.SET_INSTANCE_ID_TO_HOSTNAME_KEY, false) ? NetUtil
           .getHostnameOrAddress() : NetUtil.getHostAddress();
     }
+
     _brokerId = _brokerConf.getProperty(Helix.Instance.INSTANCE_ID_KEY,
-        Helix.PREFIX_OF_BROKER_INSTANCE + brokerHost + "_" + _brokerConf
-            .getProperty(Helix.KEY_OF_BROKER_QUERY_PORT, Helix.DEFAULT_BROKER_QUERY_PORT));
+        Helix.PREFIX_OF_BROKER_INSTANCE + brokerHost + "_" + inferPort());
```

Comment:
May be a problem here. If a broker is registered as Broker_hostxxx_8990, can it now come up as Broker_hostxxx_9661 (where 9661 is the https port)? This may mean that admins need to change the instance config for the broker, and the admins should be aware of that, right? Good to include as notes in the checkin comments, so that docs can follow that. 
Let me know if my understanding is wrong.

Reply:
We'll pick whichever port is configured first - which is usually the HTTP port, even in a multi-ingress scenario. If only HTTPS is enabled then that explicitly-configured port will be used for the instance id.
Is this what you suggest I note in the release-notes?